From: Michal Hocko <mhocko@suse.com>
Subject: mm, memory_hotplug: get rid of is_zone_device_section fix
Patch-mainline: no, part of 9d416eb0bc6e
References: bnc#1047595

Our backport of 9d416eb0bc6e ("mm, memory_hotplug: get rid of
is_zone_device_section") lacks a follow up fix noticed during the review
http://lkml.kernel.org/r/20170417201235.GA6511@redhat.com. The fix has been
folded into the patch before merging but I forgot to add it to our backport. As
a result we leave mem_sysfs_mutex locked and deadlock on any further attempt to
take this lock.

Signed-off-by: Michal Hocko <mhocko@suse.com>

---
 drivers/base/memory.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/base/memory.c
+++ b/drivers/base/memory.c
@@ -714,7 +714,7 @@ static int remove_memory_block(unsigned
 	 */
 	mem = find_memory_block(section);
 	if (!mem)
-		return 0;
+		goto out_unlock;
 
 	unregister_mem_sect_under_nodes(mem, __section_nr(section));
 
@@ -724,6 +724,7 @@ static int remove_memory_block(unsigned
 	else
 		put_device(&mem->dev);
 
+out_unlock:
 	mutex_unlock(&mem_sysfs_mutex);
 	return 0;
 }
